api/internal/target/kusttarget.go

```diff
@@ -278,6 +282,54 @@ func (kt *KustTarget) configureExternalTransformers() ([]resmap.Transformer, err
 	return kt.pLdr.LoadTransformers(kt.ldr, kt.validator, ra.ResMap())
 }
 
+func (kt *KustTarget) runValidators(ra *accumulator.ResAccumulator) error {
+	validators, err := kt.configureExternalValidators()
+	if err != nil {
+		return err
+	}
+	for _, v := range validators {
+		// Validators shouldn't modify the resource map
+		orignal := ra.ResMap().DeepCopy()
+		err = v.Validate(ra.ResMap())
+		if err != nil {
+			return err
+		}
+		new := ra.ResMap().DeepCopy()
+		kt.removeValidatedByLabel(new)
+		if err = orignal.ErrorIfNotEqualSets(new); err != nil {
+			return fmt.Errorf("validator shouldn't modify the resource map: %v", err)
+		}
+	}
+	return nil
+}
+
+func (kt *KustTarget) removeValidatedByLabel(rm resmap.ResMap) {
+	var validatedByLabelName string = "validated-by"
+
+	resources := rm.Resources()
+	for _, r := range resources {
+		labels := r.GetLabels()
+		if _, found := labels[validatedByLabelName]; !found {
+			continue
+		}
+		delete(labels, validatedByLabelName)
+		if len(labels) == 0 {
+			r.SetLabels(nil)
+		} else {
+			r.SetLabels(labels)
+		}
+	}
+}
+
+func (kt *KustTarget) configureExternalValidators() ([]resmap.Validator, error) {
```

Comment:
suggest modifying `configureExternalTransformers` to accept a field name, and re-use it to load validators and transformers.